### PR TITLE
fix serialization for SizeValues

### PR DESCRIPTION
**Pull request by vieting**
The serialization should get the `int` value of a `SizeValue`, otherwise we'll have something in the output network like `'shape': ('static_dim'(768)(768),)` which will obviously lead to an error.

### CODE REVIEW

**Comment by albertz** [comment]
Some test case which tests the serialization would be good, e.g. just serialization + the eval + dummy run.

**Comment by vieting** [comment]
> Some test case which tests the serialization would be good, e.g. just serialization + the eval + dummy run.

I added a test case. Not sure if the `arange` is the ideal test, but the serialized config contains both a dim tag and a `SizeValue`. Would you add another run based on the serialized network dict, similar to `Converter._run_returnn_standalone_net_dict`?

File: tests/test_layers.py [hunk 1]
```diff
@@ -1770,6 +1770,26 @@ def model_func(wrapped_import, inputs: torch.Tensor):
...
+    arange = torch.arange(inputs.shape[1])
+    return inputs + torch.reshape(arange, (1, -1))
+
+  rnd = numpy.random.RandomState(42)
+  x = rnd.normal(0., 1., (n_batch, n_feat)).astype("float32")
+  converter = verify_torch_and_convert_to_returnn(model_func, inputs=x)
+
+  cfg = converter.get_returnn_config_serialized()
+  print(f"Serialized config:\n\n{cfg}")
+  exec(cfg)
```
Comment (albertz) [inline tests/test_layers.py:1790]: I would further extend this to also try to load it again via RETURNN. The result of `exec` can be put into a new `cfg` object. See `config_net_dict_via_serialized` and then `dummy_run_net` in the test cases of returnn-common.

Comment (vieting) [reply tests/test_layers.py]: I added that. We have to adapt the training dataset though. Did you want to do training here also? Or just forwarding as in the other tests?

File: tests/returnn_helpers.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#     if with_axis:
+#       opts["axis"] = time_dim
+#     out = net(data, **opts, name=name_ctx)
+#     if isinstance(out, tuple):
+#       out = out[0]
+#     assert isinstance(out, nn.Tensor)
+#     out.mark_as_default_output()
+#
+#   config_code = name_ctx.get_returnn_config_serialized()
+#   return config_net_dict_via_serialized(config_code)
```
Comment (albertz) [inline tests/returnn_helpers.py:92]: You should never add commented-out code.
Instead, add a comment with a link to other code in returnn-common or so if you think this code can become relevant.

File: pytorch_to_returnn/converter/converter.py [hunk 3]
```diff
@@ -159,6 +159,7 @@ def get_returnn_config_serialized(self) -> str:
     config = dim_tags_proxy.collect_dim_tags_and_transform_config(config)
 
     code_lines = [
+      "import numpy\n",
```
Comment (albertz) [inline pytorch_to_returnn/converter/converter.py:162]: Why is this needed?

Comment (vieting) [reply pytorch_to_returnn/converter/converter.py:162]: If we have a `ConstantLayer`, it potentially contains `numpy.array[...]`.

**Comment by albertz** [comment]
Now it fails because you use `DummyDataset` which creates some data which is not compatible to what you expect here in the test, specifically you expect (B,F) but `DummyDataset` generates (T,F) and then batched as (B,T,F).

Maybe for this serialization test, just make some test which works on (B,T,F) to not make it too complicated.
